Tactical/Dialogue Control.cpp

```diff
@@ -1845,27 +1798,23 @@ void AdditionalTacticalCharacterDialogue_AllInSectorRadiusCall( UINT8 ausIgnoreP
 
 BOOLEAN CharacterDialogueWithSpecialEvent( UINT8 ubCharacterNum, UINT16 usQuoteNum, INT32 iFaceIndex, UINT8 bUIHandlerID, BOOLEAN fFromSoldier, BOOLEAN fDelayed, UINT32 uiFlag, UINT32 uiData1, UINT32 uiData2 )
 {
-	DIALOGUE_Q_STRUCT				*QItem;
-
-	// Allocate new item
-	QItem = (DIALOGUE_Q_STRUCT *) MemAlloc( sizeof( DIALOGUE_Q_STRUCT ) );
-	memset( QItem, 0, sizeof( DIALOGUE_Q_STRUCT ) );
-
-	QItem->ubCharacterNum = ubCharacterNum;
-	QItem->usQuoteNum			= usQuoteNum;
-	QItem->iFaceIndex			= iFaceIndex;
-	QItem->bUIHandlerID		= bUIHandlerID;
-	QItem->iTimeStamp			= GetJA2Clock( );
-	QItem->fFromSoldier		= fFromSoldier;
-	QItem->fDelayed				= fDelayed;
+	DIALOGUE_Q_STRUCT QItem;
+	memset(&QItem, 0, sizeof(DIALOGUE_Q_STRUCT));

```

Comment:
QItem is on the stack, do you still need to memset it?

Reply:
I was getting some really weird errors before adding these memsets to the code. I think it stemmed from QItem containing garbage data and the dialogue system then reading it when dialogue events fired. 

Things like Martha gaining 200 strength after giving enrico's letter to her, and that dialogue not starting from the correct voice line but two lines into it, etc.

Reply:
Yes, it's probably better to just zero initialize the struct's members instead of having to run memset by hand every time you want to use one, e.g.:

`DIALOGUE_Q_STRUCT QItem{};`